docs/pages/experiments/base/menu.tsx

```diff
@@ -0,0 +1,165 @@
+  margin: 2px 0;
+  border-radius: 8px;
+  cursor: default;
+  user-select: none;
+  &:last-of-type {
+    border-bottom: none;
+  }
+  &.${menuItemClasses.focusVisible} {
+    outline: 3px solid ${theme.palette.mode === 'dark' ? blue[600] : blue[200]};
+    background-color: ${theme.palette.mode === 'dark' ? grey[800] : grey[100]};
+    color: ${theme.palette.mode === 'dark' ? grey[300] : grey[900]};
+  }
+  &.${menuItemClasses.disabled} {
+    color: ${theme.palette.mode === 'dark' ? grey[700] : grey[400]};
+  }
+  &:hover:not(.${menuItemClasses.disabled}) {
+    background-color: ${theme.palette.mode === 'dark' ? grey[800] : grey[100]};
+    color: ${theme.palette.mode === 'dark' ? grey[300] : grey[900]};
+  }
+  `,
+);
+
+const StyledMenuButton = styled(MenuButton)(
+  ({ theme }) => `
+  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Noto Sans",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";;
+  font-size: 0.875rem;
+  box-sizing: border-box;
+  min-height: calc(1.5em + 22px);
+  border-radius: 12px;
+  padding: 12px 16px;
+  line-height: 1.5;
+  background: ${theme.palette.mode === 'dark' ? grey[900] : '#fff'};
+  border: 1px solid ${theme.palette.mode === 'dark' ? grey[700] : grey[200]};
+  color: ${theme.palette.mode === 'dark' ? grey[300] : grey[900]};
+  transition-property: all;
+  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
+  transition-duration: 120ms;
+  &:hover {
+    background: ${theme.palette.mode === 'dark' ? grey[800] : grey[50]};
+    border-color: ${theme.palette.mode === 'dark' ? grey[600] : grey[300]};
+  }
+  &:focus {
+    border-color: ${blue[400]};
+    outline: 3px solid ${theme.palette.mode === 'dark' ? blue[500] : blue[200]};
+  }
+  `,
+);
+
+const StyledPopper = styled(Popper)`
+  z-index: 1;
+`;
+
+const Page = styled('div')(`
+  max-width: 800px;
+  min-height: calc(100vh - 40px);
+  box-sizing: border-box;
+  margin: 20px auto;
+  padding: 20px;
+  border-radius: 4px;
+`);
+
+function UnstyledSeparator(props: React.HTMLAttributes<HTMLLIElement>) {
+  return <li {...props} role="separator" />;
+}
+
+const Separator = styled(UnstyledSeparator)(`
+  height: 0;
+  margin: 8px -6px;
+  border-bottom: 1px solid rgba(0, 0, 0, 0.1);
+`);
+
+export default function UnstyledMenuIntroduction() {
+  const [isOpen, setOpen] = React.useState(false);
+
+  const handleOpenChange = (event: React.SyntheticEvent | null, open: boolean) => {
+    setOpen(open);
+  };
+
+  const createHandleMenuClick = (menuItem: string) => {
+    return () => {
+      // eslint-disable-next-line no-console
+      console.log(`Clicked on ${menuItem}`);
+    };
+  };
+
+  return (
+    <Page>
+      <DropdownMenu open={isOpen} onOpenChange={handleOpenChange}>
+        <StyledMenuButton>Create new...</StyledMenuButton>
+        <Menu
+          slots={{ root: StyledPopper, listbox: StyledListbox }}
+          slotProps={{ root: { placement: 'bottom-start' }, listbox: { id: 'simple-menu' } }}
+        >
```

Comment:
I expect to see `aria-controls` and `id` auto-generated but there are no attributes on the experiment page:

<img width="1044" alt="image" src="https://github.com/mui/material-ui/assets/18292247/37e4d33f-6079-4c37-9ad8-ebcf4c0238e8">

Reply:
Sure thing! I haven't done the a11y part just yet